### PR TITLE
Single product page

### PR DESCRIPTION
**Pull request by yogesh-**
Added Single product page feature.
- Accessible from homepage
- Accessible from product listing page

image

### CODE REVIEW

File: src/pages/productDetails/productDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { useParams } from "react-router-dom";
+import { products } from "../../backend/db/products";
+import { Navbar } from "../../components";
+import { Footer } from "../../components";
```
Comment (shraddha15-creator) [inline src/pages/productDetails/productDetails.jsx:3]: you can write it directly in one line like this : 

` import { Navbar, Footer } from "../../components"; `

File: src/pages/productDetails/productDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import "./productDetails.css";
+import { useCart } from "../../context/cartContext";
+import { useWish } from "../../context/wishContext";
```
Comment (shraddha15-creator) [inline src/pages/productDetails/productDetails.jsx:6]: use index file in context folder, then import this from there

**Review by akash-1318 — COMMENTED** [review]
Code is really clean with proper formatting and nicely use of react concepts no changes require from my end , nice work .

File: src/pages/productDetails/productDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+export const ProductDetails = () => {
+  const { productId } = useParams();
+  const { cartDispatch } = useCart();
+  const { wishDispatch } = useWish();
+
+  const singleProduct = (productsDb, id) =>
+    productsDb.find((product) => product._id === id);
+
+  var finalSingleProduct = singleProduct(products, productId);
```
Comment (shraddha15-creator) [inline src/pages/productDetails/productDetails.jsx]: are you not calling API for a single product?

File: src/pages/products/products.css [hunk 4]
```diff
@@ -192,6 +192,20 @@ main .card-tag {
   align-items: center;
 }
 
+/* main .card-badge img:first-child {
+  width: 25rem;
+  height: 30rem;
+  margin: 2rem 2rem 0rem 2rem;
+  border-radius: 0.5rem;
+} */
+
```
Comment (shraddha15-creator) [inline src/pages/products/products.css]: remove commented code

File: src/pages/products/products.jsx [hunk 5]
```diff
@@ -59,7 +60,15 @@ export const Products = () => {
...
                 <div className="card-badge">
-                  <img src={item.image} alt="sneakers" />
+                  <Link to={`/product/${item._id}`}>
+                    <img
+                      className="img-style"
+                      src={item.image}
+                      alt="sneakers"
+                    />
+                  </Link>
+                  {/* <img src={item.image} alt="sneakers" /> */}
```
Comment (shraddha15-creator) [inline src/pages/products/products.jsx]: remove

File: src/routing.jsx [hunk 6]
```diff
@@ -6,6 +6,7 @@ import { Wishlist } from "./pages/wishlist/wishlist";
 import { Login } from "./pages/login/login";
 import { SignUp } from "./pages/signup/signup";
```
Comment (shraddha15-creator) [inline src/routing.jsx:7]: same here, use index file

**Review by jeetbhalani15 — APPROVED** [review]
Remove logs rest all looks good

**Review by shraddha15-creator — APPROVED** [review]

File: src/pages/productDetails/productDetails.jsx (hunk 3 above)
Comment (yogesh-) [reply src/pages/productDetails/productDetails.jsx]: added now

**Pull request merged**